src/qd8-f32-qb4w-gemm/gen/qd8-f32-qb4w-gemm-1x16c8-minmax-avx512vnni-prfm.c

```diff
@@ -0,0 +1,150 @@
+// Auto-generated file. Do not edit!
+//   Template: src/qs8-gemm/MRx16c8-avx512vnni.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2023 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <assert.h>
+
+#include <immintrin.h>
+
+#include "xnnpack/gemm.h"
+#include "xnnpack/intrinsics-polyfill.h"
+#include "xnnpack/math.h"
+#include "xnnpack/unaligned.h"
+#include "xnnpack/prefetch.h"
+
+
+void xnn_qd8_f32_qb4w_gemm_minmax_ukernel_1x16c8__avx512vnni_prfm(
+    size_t mr,
+    size_t nc,
+    size_t kc,
+    const int8_t* restrict a,
+    size_t a_stride,
+    const void* restrict w,
+    float* restrict c,
+    size_t cm_stride,
+    size_t cn_stride,
+    const union xnn_f32_qb4w_minmax_params params[restrict XNN_MIN_ELEMENTS(1)],
+    const struct xnn_qd8_quantization_params quantization_params[restrict XNN_MIN_ELEMENTS(1)]) XNN_OOB_READS
+{
+  assert(mr != 0);
+  assert(mr <= 1);
+  assert(nc != 0);
+  assert(kc != 0);
+  assert(kc % sizeof(int8_t) == 0);
+  assert(a != NULL);
+  assert(w != NULL);
+  assert(c != NULL);
+
+  kc = round_up_po2(kc, 8 * sizeof(int8_t));
+  const int8_t* a0 = a;
+  float* c0 = c;
+  size_t bl = params->avx512vnni.blocksize;
+  assert(bl != 0);
+  assert(bl <= kc);
+  size_t n_blocks = kc / bl;
+
+  const __m512 vinput_zero_point0 = _mm512_set1_ps((float) quantization_params[0].zero_point + 128);
+  const __m512 voutput_min = _mm512_set1_ps(params->avx512vnni.min);
+  const __m512 voutput_max = _mm512_set1_ps(params->avx512vnni.max);
+  const __m512i vsign_mask = _mm512_set1_epi8(params->avx512vnni.sign_mask);  // 0x80
+  const __m512i vvalue_mask = _mm512_set1_epi8(params->avx512vnni.mask);  // 0xF0
+  do {
+    const __m512 vksum0123456789ABCDEF = _mm512_load_ps(w);
+    __m512 vscaled0x0123456789ABCDEF = _mm512_mul_ps(vksum0123456789ABCDEF, vinput_zero_point0);
+    w = (const int32_t*) w + 16;
+
+    for (size_t nb=0; nb < n_blocks; ++nb) {
```

Comment:
instead of a block count, use kc
something like:
for (size_t k=0; k < kc; k += bl) {
remove n_blocks

Reply:
done. I did it this way for readability.